File: be/SpringBoot/src/main/java/com/common/dto/response/UserDTO.java
package com.common.dto.response;

import com.common.model.enumerate.EntityStatus;
import com.common.model.enumerate.UserRole;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;

import java.time.LocalDateTime;
import java.util.Set;

/**
 * DTO đại diện cho dữ liệu User.
 *
 * @created_at 2026-07-31
 * @author txhoan
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class UserDTO {

    Long id;

    String username;

    String email;

    EntityStatus status;

    Long tenantId;

    String tenantCode;

    Set<UserRole> roles;

    ProfileDTO profile;

    LocalDateTime createdAt;

    LocalDateTime modifiedAt;
}

File: be/.Net/Swe.BL/DTO/Response/UserDTO.cs
using System;
using System.Collections.Generic;

namespace Swe.BL.DTO.Response;

/// <summary>
/// DTO đại diện cho dữ liệu User.
/// </summary>
/// <created_at>2026-08-03</created_at>
/// <author>txhoan</author>
public class UserDTO
{
    public long Id { get; set; }
    public string Username { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Status { get; set; } = string.Empty;
    public long? TenantId { get; set; }
    public string? TenantCode { get; set; }
    public List<string> Roles { get; set; } = new();
    public ProfileDTO? Profile { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? ModifiedAt { get; set; }
}
